Sources/UnidirectionalBinding.swift

```diff
@@ -304,14 +305,17 @@ public final class BindingTarget<Value>: BindingTargetProtocol {
 	///   - setter: The action to receive values.
 	public convenience init(on queue: DispatchQueue, lifetime: Lifetime, setter: @escaping (Value) -> Void) {
 		let queueId = ObjectIdentifier(queue)
+		let usesMainQueue = queue == DispatchQueue.main
 
 		/// Ensures the queue has been setup property.
 		if nil == queue.getSpecific(key: specificKey) {
 			queue.setSpecific(key: specificKey, value: queueId)
 		}
 
 		let setter: (Value) -> Void = { value in
-			if queueId == DispatchQueue.getSpecific(key: specificKey) {
+			if usesMainQueue && Thread.isMainThread {
```

Comment:
`Scheduler` is asynchronous. Using it would make all Cocoa binding targets to accept values asynchronously.